### ISSUE TITLE
Integrity constraint errors loading sample data in new db

### ISSUE THREAD

**Issue by specshack** [magento/magento2#1308]
Using the CLI instruction
>php dev/tools/Magento/Tools/SampleData/install.php --admin_user=Admin

to load the data for **0.74.0-beta10**, I am receiving the following error when it begins to load Orders:

>Installing orders:
.exception 'PDOException' with message 'SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '000000001-1' for key 'UNQ_SALES_ORDER_INCREMENT_ID_STORE_ID'' in /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php:228

**Stack trace looks like:**

Stack trace:
#0 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php(228): PDOStatement->execute(Array)
#1 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/DB/Statement/Pdo/Mysql.php(95): Zend_Db_Statement_Pdo->_execute(Array)
#2 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Statement.php(303): Magento\Framework\DB\Statement\Pdo\Mysql->_execute(Array)
#3 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#4 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query('INSERT INTO `sa...', Array)
#5 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php(428): Zend_Db_Adapter_Pdo_Abstract->query('INSERT INTO `sa...', Array)
#6 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(576): Magento\Framework\DB\Adapter\Pdo\Mysql->query('INSERT INTO `sa...', Array)
#7 /Library/WebServer/Documents/magento2/app/code/Magento/Sales/Model/Resource/EntityAbstract.php(240): Zend_Db_Adapter_Abstract->insert('sales_order', Array)
#8 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Model/Resource/Order/Interceptor.php(148): Magento\Sales\Model\Resource\EntityAbstract->save(Object(Magento\Sales\Model\Order))
#9 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/OrderInterfacePersistor.php(146): Magento\Sales\Model\Resource\Order\Interceptor->save(Object(Magento\Sales\Model\Order))
#10 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/OrderInterfacePersistor.php(180): Magento\Sales\Api\Data\OrderInterfacePersistor->doPersist(1)
#11 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository.php(162): Magento\Sales\Api\Data\OrderInterfacePersistor->doPersistEntity(Object(Magento\Sales\Model\Order))
#12 [internal function]: Magento\Sales\Api\Data\Order\Repository->save(Object(Magento\Sales\Model\Order))
#13 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(56): call_user_func_array(Array, Array)
#14 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/Interception/Chain/Chain.php(65): Magento\Sales\Api\Data\Order\Repository\Interceptor->___callParent('save', Array)
#15 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(95): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Sales\\A...', 'save', Object(Magento\Sales\Api\Data\Order\Repository\Interceptor), Array, 'save_gift_messa...')
#16 /Library/WebServer/Documents/magento2/app/code/Magento/GiftMessage/Model/Plugin/OrderSave.php(49): Magento\Sales\Api\Data\Order\Repository\Interceptor->Magento\Sales\Api\Data\Order\Repository\{closure}(Object(Magento\Sales\Model\Order))
#17 [internal function]: Magento\GiftMessage\Model\Plugin\OrderSave->aroundSave(Object(Magento\Sales\Api\Data\Order\Repository\Interceptor), Object(Closure), Object(Magento\Sales\Model\Order))
#18 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(100): call_user_func_array(Array, Array)
#19 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(228): Magento\Sales\Api\Data\Order\Repository\Interceptor->___callPlugins('save', Array, Array)
#20 /Library/WebServer/Documents/magento2/app/code/Magento/Sales/Model/Service/OrderService.php(160): Magento\Sales\Api\Data\Order\Repository\Interceptor->save(Object(Magento\Sales\Model\Order))
#21 /Library/WebServer/Documents/magento2/app/code/Magento/Quote/Model/QuoteManagement.php(468): Magento\Sales\Model\Service\OrderService->place(Object(Magento\Sales\Model\Order))
#22 /Library/WebServer/Documents/magento2/app/code/Magento/Quote/Model/QuoteManagement.php(368): Magento\Quote\Model\QuoteManagement->submitQuote(Object(Magento\Quote\Model\Quote), Array)
#23 /Library/WebServer/Documents/magento2/app/code/Magento/Sales/Model/AdminOrder/Create.php(1855): Magento\Quote\Model\QuoteManagement->submit(Object(Magento\Quote\Model\Quote), Array)
#24 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Module/Sales/Setup/Order/Processor.php(136): Magento\Sales\Model\AdminOrder\Create->createOrder()
#25 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Module/Sales/Setup/Order.php(77): Magento\Tools\SampleData\Module\Sales\Setup\Order\Processor->createOrder(Array)
#26 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Helper/PostInstaller.php(86): Magento\Tools\SampleData\Module\Sales\Setup\Order->run()
#27 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Installer.php(99): Magento\Tools\SampleData\Helper\PostInstaller->run()
#28 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/ConsoleInstaller.php(78): Magento\Tools\SampleData\Installer->run('Admin', Array)
#29 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/App/Bootstrap.php(257): Magento\Tools\SampleData\ConsoleInstaller->launch()
#30 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/install.php(31): Magento\Framework\App\Bootstrap->run(Object(Magento\Tools\SampleData\ConsoleInstaller))
#31 {main}

**Which is immediately followed by another error:**

Next exception 'Zend_Db_Statement_Exception' with message 'SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '000000001-1' for key 'UNQ_SALES_ORDER_INCREMENT_ID_STORE_ID', query was: INSERT INTO `sales_order` (`state`, `status`, `coupon_code`, `protect_code`, `shipping_description`, `is_virtual`, `store_id`, `customer_id`, `base_discount_amount`, `base_discount_canceled`, `base_discount_invoiced`, `base_discount_refunded`, `base_grand_total`, `base_shipping_amount`, `base_shipping_canceled`, `base_shipping_invoiced`, `base_shipping_refunded`, `base_shipping_tax_amount`, `base_shipping_tax_refunded`, `base_subtotal`, `base_subtotal_canceled`, `base_subtotal_invoiced`, `base_subtotal_refunded`, `base_tax_amount`, `base_tax_canceled`, `base_tax_invoiced`, `base_tax_refunded`, `base_to_global_rate`, `base_to_order_rate`, `base_total_canceled`, `base_total_invoiced`, `base_total_invoiced_cost`, `base_total_offline_refunded`, `base_total_online_refunded`, `base_total_paid`, `base_total_qty_ordered`, `base_total_refunded`, `discount_amount`, `discount_canceled`, `discount_invoiced`, `discount_refunded`, `grand_total`, `shipping_amount`, `shipping_canceled`, `shipping_invoiced`, `shipping_refunded`, `shipping_tax_amount`, `shipping_tax_refunded`, `store_to_base_rate`, `store_to_order_rate`, `subtotal`, `subtotal_canceled`, `subtotal_invoiced`, `subtotal_refunded`, `tax_amount`, `tax_canceled`, `tax_invoiced`, `tax_refunded`, `total_canceled`, `total_invoiced`, `total_offline_refunded`, `total_online_refunded`, `total_paid`, `total_qty_ordered`, `total_refunded`, `can_ship_partially`, `can_ship_partially_item`, `customer_is_guest`, `customer_note_notify`, `billing_address_id`, `customer_group_id`, `edit_increment`, `email_sent`, `forced_shipment_with_invoice`, `payment_auth_expiration`, `quote_address_id`, `quote_id`, `shipping_address_id`, `adjustment_negative`, `adjustment_positive`, `base_adjustment_negative`, `base_adjustment_positive`, `base_shipping_discount_amount`, `base_subtotal_incl_tax`, `base_total_due`, `payment_authorization_amount`, `shipping_discount_amount`, `subtotal_incl_tax`, `total_due`, `weight`, `customer_dob`, `increment_id`, `applied_rule_ids`, `base_currency_code`, `customer_email`, `customer_firstname`, `customer_lastname`, `customer_middlename`, `customer_prefix`, `customer_suffix`, `customer_taxvat`, `discount_description`, `ext_customer_id`, `ext_order_id`, `global_currency_code`, `hold_before_state`, `hold_before_status`, `order_currency_code`, `original_increment_id`, `relation_child_id`, `relation_child_real_id`, `relation_parent_id`, `relation_parent_real_id`, `remote_ip`, `shipping_method`, `store_currency_code`, `store_name`, `x_forwarded_for`, `customer_note`, `updated_at`, `total_item_count`, `customer_gender`, `hidden_tax_amount`, `base_hidden_tax_amount`, `shipping_hidden_tax_amount`, `base_shipping_hidden_tax_amnt`, `hidden_tax_invoiced`, `base_hidden_tax_invoiced`, `hidden_tax_refunded`, `base_hidden_tax_refunded`, `shipping_incl_tax`, `base_shipping_incl_tax`, `gift_message_id`) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, NULL, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' in /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php:235

**And its stack trace:**

Stack trace:
#0 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/DB/Statement/Pdo/Mysql.php(95): Zend_Db_Statement_Pdo->_execute(Array)
#1 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Statement.php(303): Magento\Framework\DB\Statement\Pdo\Mysql->_execute(Array)
#2 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#3 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query('INSERT INTO `sa...', Array)
#4 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php(428): Zend_Db_Adapter_Pdo_Abstract->query('INSERT INTO `sa...', Array)
#5 /Library/WebServer/Documents/magento2/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(576): Magento\Framework\DB\Adapter\Pdo\Mysql->query('INSERT INTO `sa...', Array)
#6 /Library/WebServer/Documents/magento2/app/code/Magento/Sales/Model/Resource/EntityAbstract.php(240): Zend_Db_Adapter_Abstract->insert('sales_order', Array)
#7 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Model/Resource/Order/Interceptor.php(148): Magento\Sales\Model\Resource\EntityAbstract->save(Object(Magento\Sales\Model\Order))
#8 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/OrderInterfacePersistor.php(146): Magento\Sales\Model\Resource\Order\Interceptor->save(Object(Magento\Sales\Model\Order))
#9 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/OrderInterfacePersistor.php(180): Magento\Sales\Api\Data\OrderInterfacePersistor->doPersist(1)
#10 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository.php(162): Magento\Sales\Api\Data\OrderInterfacePersistor->doPersistEntity(Object(Magento\Sales\Model\Order))
#11 [internal function]: Magento\Sales\Api\Data\Order\Repository->save(Object(Magento\Sales\Model\Order))
#12 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(56): call_user_func_array(Array, Array)
#13 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/Interception/Chain/Chain.php(65): Magento\Sales\Api\Data\Order\Repository\Interceptor->___callParent('save', Array)
#14 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(95): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Sales\\A...', 'save', Object(Magento\Sales\Api\Data\Order\Repository\Interceptor), Array, 'save_gift_messa...')
#15 /Library/WebServer/Documents/magento2/app/code/Magento/GiftMessage/Model/Plugin/OrderSave.php(49): Magento\Sales\Api\Data\Order\Repository\Interceptor->Magento\Sales\Api\Data\Order\Repository\{closure}(Object(Magento\Sales\Model\Order))
#16 [internal function]: Magento\GiftMessage\Model\Plugin\OrderSave->aroundSave(Object(Magento\Sales\Api\Data\Order\Repository\Interceptor), Object(Closure), Object(Magento\Sales\Model\Order))
#17 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(100): call_user_func_array(Array, Array)
#18 /Library/WebServer/Documents/magento2/var/generation/Magento/Sales/Api/Data/Order/Repository/Interceptor.php(228): Magento\Sales\Api\Data\Order\Repository\Interceptor->___callPlugins('save', Array, Array)
#19 /Library/WebServer/Documents/magento2/app/code/Magento/Sales/Model/Service/OrderService.php(160): Magento\Sales\Api\Data\Order\Repository\Interceptor->save(Object(Magento\Sales\Model\Order))
#20 /Library/WebServer/Documents/magento2/app/code/Magento/Quote/Model/QuoteManagement.php(468): Magento\Sales\Model\Service\OrderService->place(Object(Magento\Sales\Model\Order))
#21 /Library/WebServer/Documents/magento2/app/code/Magento/Quote/Model/QuoteManagement.php(368): Magento\Quote\Model\QuoteManagement->submitQuote(Object(Magento\Quote\Model\Quote), Array)
#22 /Library/WebServer/Documents/magento2/app/code/Magento/Sales/Model/AdminOrder/Create.php(1855): Magento\Quote\Model\QuoteManagement->submit(Object(Magento\Quote\Model\Quote), Array)
#23 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Module/Sales/Setup/Order/Processor.php(136): Magento\Sales\Model\AdminOrder\Create->createOrder()
#24 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Module/Sales/Setup/Order.php(77): Magento\Tools\SampleData\Module\Sales\Setup\Order\Processor->createOrder(Array)
#25 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Helper/PostInstaller.php(86): Magento\Tools\SampleData\Module\Sales\Setup\Order->run()
#26 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/Installer.php(99): Magento\Tools\SampleData\Helper\PostInstaller->run()
#27 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/ConsoleInstaller.php(78): Magento\Tools\SampleData\Installer->run('Admin', Array)
#28 /Library/WebServer/Documents/magento2/lib/internal/Magento/Framework/App/Bootstrap.php(257): Magento\Tools\SampleData\ConsoleInstaller->launch()
#29 /Library/WebServer/Documents/magento2/dev/tools/Magento/Tools/SampleData/install.php(31): Magento\Framework\App\Bootstrap->run(Object(Magento\Tools\SampleData\ConsoleInstaller))
#30 {main}Daves-MacBook-Pro:magento2 dave$ 
  [Restored]

**Comment by ddeppner** [comment]
I followed the precise instructions for installing rgranadino's Magento 2 vagrant box at:

https://github.com/rgranadino/mage2_vagrant

At step 6 of his install instructions I chose "reinstall -s" to install sample data and reproduced this error exactly:

...
Installing orders:
.exception 'PDOException' with message 'SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '000000001-1' for key 'UNQ_SALES_ORDER_INCREMENT_ID_STORE_ID'' in /vagrant/data/magento2/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php:228
...

**Comment by buskamuza** [comment]
@specshack , @ddeppner , please, try it now. Updated structure of Sample Data has been just published and it should be installed correctly (Sample Data packages of version `0.42.0-beta10` should be fetched by Composer)
Tested on this Vagrant installation https://github.com/buskamuza/magento2-product-vagrant

**Comment by specshack** [comment]
Getting completely different errors now BEFORE even getting to the install data steps:

In this section of the install process:

>Installing user configuration...

the entire installation stalls at this point:

>Module 'Magento_Catalog':
Installing data.. 

This is at approximately the 70% completed mark. 

This is happening with a fresh clone. No errors are reported in the logs. The install log shows only the following status statement:

>class="text-info">Module 'Magento_Catalog':</span><br><span class="text-info">Installing data.. </span> 

The debug log shows:

>[2015-05-26 23:10:10] main.DEBUG: cache_invalidate:  {"method":"POST","url":"http://localhost/magento2/setup/index.php/install/start","invalidateInfo":{"tags":["EAV"],"mode":"matchingAnyTag"},"is_exception":false} []

**Comment by specshack** [comment]
FWIW, I am not running this on a Vagrant box.

**Comment by buskamuza** [comment]
@specshack, thanks for the info. Could you, please, clarify couple more things?
1. Do you run installation on an empty db?
2. Could you, please, provide full installation output?

Thanks

**Comment by adzminz** [comment]
@specshack @buskamuza, I got the same error while installing Magento2 via web browser. The installation process stopped around 70%. 

I ran installation on an empty db.

**Comment by specshack** [comment]
Yes, it ran on an empty db.

Installation output:

Starting Magento installation:
File permissions check...

Enabling Maintenance Mode...

Installing deployment configuration...

Installing database schema:
Schema creation/updates:
Module 'Magento_Store':
Installing schema.. 

Module 'Magento_Directory':
Installing schema.. 

Module 'Magento_Backend':

Module 'Magento_Backup':

Module 'Magento_Theme':
Installing schema.. Upgrading schema.. 

Module 'Magento_CacheInvalidate':

Module 'Magento_Eav':
Installing schema.. 

Module 'Magento_Customer':
Installing schema.. Upgrading schema.. 

Module 'Magento_CatalogImportExport':

Module 'Magento_Indexer':
Installing schema.. 

Module 'Magento_Cms':
Installing schema.. 

Module 'Magento_Search':
Installing schema.. Upgrading schema.. 

Module 'Magento_Catalog':
Installing schema.. Upgrading schema.. 

Module 'Magento_Rule':

Module 'Magento_CatalogInventory':
Installing schema.. 

Module 'Magento_Payment':

Module 'Magento_CheckoutAgreements':
Installing schema.. 

Module 'Magento_Quote':
Installing schema.. 

Module 'Magento_CmsUrlRewrite':

Module 'Magento_Config':
Installing schema.. 

Module 'Magento_ConfigurableImportExport':

Module 'Magento_Msrp':

Module 'Magento_Contact':

Module 'Magento_Cookie':

Module 'Magento_Cron':
Installing schema.. 

Module 'Magento_Widget':
Installing schema.. 

Module 'Magento_Bundle':
Installing schema.. 

Module 'Magento_CustomerImportExport':

Module 'Magento_DesignEditor':
Installing schema.. 

Module 'Magento_Developer':

Module 'Magento_Dhl':

Module 'Magento_Authorization':
Installing schema.. 

Module 'Magento_Downloadable':
Installing schema.. 

Module 'Magento_SalesSequence':
Installing schema.. 

Module 'Magento_Email':
Installing schema.. 

Module 'Magento_Fedex':

Module 'Magento_Sales':
Installing schema.. Upgrading schema.. 

Module 'Magento_Checkout':

Module 'Magento_GoogleAnalytics':

Module 'Magento_GoogleOptimizer':
Installing schema.. 

Module 'Magento_GoogleShopping':
Installing schema.. 

Module 'Magento_GroupedImportExport':

Module 'Magento_GroupedProduct':

Module 'Magento_ImportExport':
Installing schema.. 

Module 'Magento_CatalogRule':
Installing schema.. 

Module 'Magento_User':
Installing schema.. 

Module 'Magento_LayeredNavigation':

Module 'Magento_Log':
Installing schema.. 

Module 'Magento_MediaStorage':

Module 'Magento_ConfigurableProduct':
Installing schema.. 

Module 'Magento_Multishipping':

Module 'Magento_Newsletter':
Installing schema.. 

Module 'Magento_OfflinePayments':

Module 'Magento_SalesRule':
Installing schema.. 

Module 'Magento_PageCache':

Module 'Magento_Centinel':

Module 'Magento_Persistent':
Installing schema.. 

Module 'Magento_ProductAlert':
Installing schema.. 

Module 'Magento_Captcha':
Installing schema.. 

Module 'Magento_Reports':
Installing schema.. 

Module 'Magento_RequireJs':

Module 'Magento_Review':
Installing schema.. 

Module 'Magento_Rss':

Module 'Magento_CatalogWidget':

Module 'Magento_GoogleAdwords':

Module 'Magento_OfflineShipping':
Installing schema.. 

Module 'Magento_GiftMessage':
Installing schema.. 

Module 'Magento_SampleData':

Module 'Magento_SampleInterception':

Module 'Magento_SampleMinimal':

Module 'Magento_SampleModifyContent':

Module 'Magento_SampleNewPage':

Module 'Magento_SampleServiceContractClient':

Module 'Magento_SampleServiceContractReplacement':

Module 'Magento_SampleWebFlow':

Module 'Magento_SampleWebapiClient':

Module 'Magento_CatalogSearch':
Installing schema.. 

Module 'Magento_Sendfriend':
Installing schema.. 

Module 'Magento_Shipping':

Module 'Magento_Sitemap':
Installing schema.. 

Module 'Magento_AdminNotification':
Installing schema.. 

Module 'Magento_Tax':
Installing schema.. 

Module 'Magento_TaxImportExport':

Module 'Magento_CatalogUrlRewrite':
Installing schema.. 

Module 'Magento_Translation':
Installing schema.. 

Module 'Magento_Ui':

Module 'Magento_Ups':

Module 'Magento_UrlRewrite':
Installing schema.. 

Module 'Magento_Integration':
Installing schema.. 

Module 'Magento_Usps':

Module 'Magento_Variable':
Installing schema.. 

Module 'Magento_Version':

Module 'Magento_Webapi':

Module 'Magento_Weee':
Installing schema.. 

Module 'Magento_CurrencySymbol':

Module 'Magento_Wishlist':
Installing schema.. 

Schema post-updates:
Module 'Magento_Store':

Module 'Magento_Directory':

Module 'Magento_Backend':

Module 'Magento_Backup':

Module 'Magento_Theme':

Module 'Magento_CacheInvalidate':

Module 'Magento_Eav':

Module 'Magento_Customer':

Module 'Magento_CatalogImportExport':

Module 'Magento_Indexer':

Module 'Magento_Cms':

Module 'Magento_Search':

Module 'Magento_Catalog':

Module 'Magento_Rule':

Module 'Magento_CatalogInventory':

Module 'Magento_Payment':

Module 'Magento_CheckoutAgreements':

Module 'Magento_Quote':

Module 'Magento_CmsUrlRewrite':

Module 'Magento_Config':

Module 'Magento_ConfigurableImportExport':

Module 'Magento_Msrp':

Module 'Magento_Contact':

Module 'Magento_Cookie':

Module 'Magento_Cron':

Module 'Magento_Widget':

Module 'Magento_Bundle':

Module 'Magento_CustomerImportExport':

Module 'Magento_DesignEditor':

Module 'Magento_Developer':

Module 'Magento_Dhl':

Module 'Magento_Authorization':

Module 'Magento_Downloadable':

Module 'Magento_SalesSequence':

Module 'Magento_Email':

Module 'Magento_Fedex':

Module 'Magento_Sales':

Module 'Magento_Checkout':

Module 'Magento_GoogleAnalytics':

Module 'Magento_GoogleOptimizer':

Module 'Magento_GoogleShopping':

Module 'Magento_GroupedImportExport':

Module 'Magento_GroupedProduct':

Module 'Magento_ImportExport':

Module 'Magento_CatalogRule':

Module 'Magento_User':

Module 'Magento_LayeredNavigation':

Module 'Magento_Log':

Module 'Magento_MediaStorage':

Module 'Magento_ConfigurableProduct':

Module 'Magento_Multishipping':

Module 'Magento_Newsletter':

Module 'Magento_OfflinePayments':

Module 'Magento_SalesRule':

Module 'Magento_PageCache':

Module 'Magento_Centinel':

Module 'Magento_Persistent':

Module 'Magento_ProductAlert':

Module 'Magento_Captcha':

Module 'Magento_Reports':

Module 'Magento_RequireJs':

Module 'Magento_Review':

Module 'Magento_Rss':

Module 'Magento_CatalogWidget':

Module 'Magento_GoogleAdwords':

Module 'Magento_OfflineShipping':

Module 'Magento_GiftMessage':

Module 'Magento_SampleData':

Module 'Magento_SampleInterception':

Module 'Magento_SampleMinimal':

Module 'Magento_SampleModifyContent':

Module 'Magento_SampleNewPage':

Module 'Magento_SampleServiceContractClient':

Module 'Magento_SampleServiceContractReplacement':

Module 'Magento_SampleWebFlow':

Module 'Magento_SampleWebapiClient':

Module 'Magento_CatalogSearch':

Module 'Magento_Sendfriend':

Module 'Magento_Shipping':

Module 'Magento_Sitemap':

Module 'Magento_AdminNotification':

Module 'Magento_Tax':

Module 'Magento_TaxImportExport':

Module 'Magento_CatalogUrlRewrite':

Module 'Magento_Translation':

Module 'Magento_Ui':

Module 'Magento_Ups':

Module 'Magento_UrlRewrite':

Module 'Magento_Integration':

Module 'Magento_Usps':

Module 'Magento_Variable':

Module 'Magento_Version':

Module 'Magento_Webapi':

Module 'Magento_Weee':

Module 'Magento_CurrencySymbol':

Module 'Magento_Wishlist':


Installing user configuration...

Installing data...
Data install/update:
Module 'Magento_Store':

Module 'Magento_Directory':
Installing data.. 

Module 'Magento_Backend':

Module 'Magento_Backup':

Module 'Magento_Theme':
Installing data.. 

Module 'Magento_CacheInvalidate':

Module 'Magento_Eav':
Installing data.. 

Module 'Magento_Customer':
Installing data.. 

Module 'Magento_CatalogImportExport':

Module 'Magento_Indexer':

Module 'Magento_Cms':
Installing data.. 

Module 'Magento_Search':

Module 'Magento_Catalog':
Installing data..

**Comment by mazhalai** [comment]
@specshack can you please try `bin/magento setup:uninstall` before installing again?

**Comment by specshack** [comment]
I ran that before this install attempt. I will do it again and report back

**Comment by specshack** [comment]
After running **magento setup:uninstall** and then **composer update** the installation via the web interface completed, including installation of sample data seemingly without error.

>Seemingly without error.

Upon completion of the install, I clicked on the **Launch Admin** button and immediately got a 404:

 >The requested URL /magento2/admin/ was not found on this server.

I then checked the frontend and got something of a hot mess of a page. The topmost section shows only unformatted data and links. The middle section of the page appears properly formatted showing products, photos, etc., via the luma theme as expected. The bottom section of the page again shows only unformatted data and links again.

Next step, I verified all permissions on directories and files under /magento2, I deleted all cached files under /var, restarted the apache2 server and got exactly the same result.

So, in summary, I seem to have a properly built database loaded with sample data now (which is the problem I started with), but now don't have access to the backend and have a flaky frontend.

Any advice would be helpful.

**Comment by specshack** [comment]
Nope. System stops again at 70% installation threshold (see prior install log in earlier message, it's identical). 

Am not able to install the system at all, even without sample data. No idea what's going on now. I've been successfully installing since the -alphas and have never had this problem. Getting very frustrated and wishing I had skipped this rev.

**Comment by buskamuza** [comment]
@specshack , I'm sorry that you have so bad experience with installation.
Could you, please, try CLI installation? 
```
./bin/magento setup:install --db-host=localhost --db-name=magento --db-user=magento --db-password=magento --base-url=http://magento2.lh/ --backend-frontname=admin --admin-firstname=Olga --admin-lastname=Kopylova --admin-email=admin@test.com --admin-user=admin --admin-password=1amtheadmin
``` 
(substitute with your values).

Web Installation doesn't handle errors correctly in some cases on some environments and CLI version is more verbose (fixes are not in the repository yet).

Do I understand correctly, that you could install Magento via Web Wizard with Sample Data once? And after that it doesn't work again.

**Comment by specshack** [comment]
**Here's the new status:**

AFTER I deleted my local repo, I cloned the latest rev directly from github via the CLI. I had previously been cloning to the desktop (it's always worked in the past). I cleared the composer cache, re-ran composer install with sample data enabled in the composer.json file. As the following shows, composer seems to have done its job properly (these are the last two lines, but the rest all ran correctly:

>  - Installing magento/sample-data-media (0.42.0-beta2)
    Downloading: 100%         

>  - Installing magento/module-sample-data (0.74.0-beta10)
    Downloading: 100%         

>Writing lock file
>Generating autoload files

**Following that I ran the following command:**

>Daves-MacBook-Pro:magento2 dave$ ./bin/magento setup:install --db-host=localhost --db-name=magento2 --db-user=root --db-password=root --base-url=http://localhost/magento2/ --backend-frontname=admin --admin-firstname=Dave --admin-lastname=Garbe --admin-email=dgarbe@mac.com --admin-user=Admin --admin-password=Tortuga--21
 
The installer ran, but never showed an attempt to install the sample-data, as shown here:

>Starting Magento installation:
File permissions check...
[Progress: 1 / 274]
Enabling Maintenance Mode...
[Progress: 2 / 274]

...

**and finishing with:**

>Module 'Magento_CurrencySymbol':
[Progress: 268 / 274]
Module 'Magento_Wishlist':
[Progress: 269 / 274]
[Progress: 270 / 274]
Installing admin user...
[Progress: 271 / 274]
Enabling caches:
Current status:
Array
(
    [config] => 1
    [layout] => 1
    [block_html] => 1
    [view_files_fallback] => 1
    [view_files_preprocessing] => 1
    [collections] => 1
    [db_ddl] => 1
    [eav] => 1
    [full_page] => 1
    [translate] => 1
    [config_integration] => 1
    [config_integration_api] => 1
    [config_webservice] => 1
)

>[Progress: 272 / 274]
Disabling Maintenance Mode:
[Progress: 273 / 274]
Post installation file permissions check...
For security, remove write permissions from these directories: '/Library/WebServer/Documents/magento2/app/etc'
[Progress: 274 / 274]
[SUCCESS]: Magento installation complete.

I thought it was odd that it showed the sample-data and sample-data-media files a being downloaded by composer so I looked in the file system under:

>/dev/tools/Magento/Tools

and lo and behold there is no sample data

>Daves-MacBook-Pro:Tools dave$ ls -l
total 0
drwx------   3 dave  admin  102 May 27 19:49 Console
drwx------   3 dave  admin  102 May 27 19:49 Layout
drwx------  13 dave  admin  442 May 27 19:49 Migration
drwx------   4 dave  admin  136 May 27 19:49 StaticReview
drwx------   5 dave  admin  170 May 27 19:49 classmap
drwx------   3 dave  admin  102 May 27 19:49 psr

Any thoughts? Thankfully the system itself DOES install and both front and backends are working properly now. 

I have no idea why composer is showing the sample data as 100% downloaded but the files themselves are nowhere to be found.

**Comment by specshack** [comment]
Finally got everything installed and working via web interface. Never did get the CLI to play nice re: loading the sample data. Always failed to find the input file.

This thread can go silent now.

**Comment by mazhalai** [comment]
Thank you @specshack for confirming that it is working now. Please continue to report issues if you come across them.

**Comment by buskamuza** [comment]
@specshack , just few notes:

1. I see you use "localhost" as a domain. In some browsers it may lead to issues with cookies (because it's not considered a normal domain), so it's better to use a domain with dots (e.g., magento.lh or 127.0.0.1)

2. to install Sample Data during installation of Magento, use `--use-sample-data` option for CLI command. It's not installed by default.

3. Sample Data consists of 2 packages: SampleData module, which is installed into `app/code/Magento/SampleData`, and a media package, which remains in `vendor`. That's why you could not find it in `dev/tools`
